lib/pbench/server/api/resources/upload_api.py

```diff
@@ -50,70 +51,90 @@ def __init__(self, config, logger):
 
-        try:
-            content_length = int(request.headers.get("Content-Length"))
-        except ValueError:
-            self.logger.debug(
-                f"Tarfile upload: Invalid content-length header, not an integer for file {filename}"
-            )
-            abort(400, message="Invalid content-length header, not an integer")
-        except Exception:
-            self.logger.debug(
-                f"Tarfile upload: No Content-Length header value found for file {filename}"
-            )
-            abort(400, message="Missing required content-length header")
+        msg = ""
+        status = None
+        content_length_hdr = request.headers.get("Content-Length")
+        if not content_length_hdr:
+            msg = "Missing required Content-Length header"
+            status = HTTPStatus.LENGTH_REQUIRED
         else:
-            if content_length > self.max_content_length:
-                self.logger.debug(
-                    f"Tarfile upload: Content-Length exceeded maximum upload size allowed. File: {filename}"
-                )
-                abort(
-                    400,
-                    message=f"Payload body too large, {content_length:d} bytes, maximum size should be less than "
-                    f"or equal to {humanize.naturalsize(self.max_content_length)}",
-                )
-            elif content_length == 0:
-                self.logger.debug(
-                    f"Tarfile upload: Content-Length header value is 0 for file {filename}"
-                )
-                abort(
-                    400,
-                    message="Upload failed, Content-Length received in header is 0",
-                )
+            try:
+                content_length = int(content_length_hdr)
+            except ValueError:
+                msg = f"Invalid Content-Length header, not an integer ({content_length_hdr})"
+                status = HTTPStatus.BAD_REQUEST
+            else:
+                if content_length > self.max_content_length:
+                    cl = humanize.naturalsize(content_length)
+                    mcl = humanize.naturalsize(self.max_content_length)
+                    msg = f"Content-Length greater than maximum allowed size ({cl} > {mcl})"
+                    status = HTTPStatus.REQUEST_ENTITY_TOO_LARGE
+                elif content_length == 0:
+                    msg = "Invalid Content-Length header, zero (0)"
+                    status = HTTPStatus.BAD_REQUEST
+                elif content_length < 0:
+                    msg = f"Invalid Content-Length header, negative ({content_length})"
+                    status = HTTPStatus.BAD_REQUEST
+        if status:
+            self.logger.warning(
+                "{} for user = {}, ctrl = {!a}, file = {!a}",
+                msg,
+                username,
+                controller,
+                filename,
+            )
+            abort(status, message=msg)
```

Comment:
How about this?
```
        status = HTTPStatus.OK
        try:
            content_length = int(request.headers["Content-Length"])
        except KeyError:
            msg = "Missing required Content-Length header"
            status = HTTPStatus.LENGTH_REQUIRED
        except ValueError:
            msg = f"Invalid Content-Length header, not an integer ({content_length_hdr})"
            status = HTTPStatus.BAD_REQUEST
        else:
            if not (0 < content_length <= self.max_content_length):
                msg = ("Content-Length ({}) must be greater than 0 and no greater than {}"
                        .format(content_length, humanize.naturalsize(self.max_content_length))
                status = (HTTPStatus.REQUEST_ENTITY_TOO_LARGE
                        if 0 < content_length else HTTPStatus.BAD_REQUEST)
        if status != HTTPStatus.OK:
            self.logger.warning(
                "{} for user = {}, ctrl = {!a}, file = {!a}",
                msg,
                username,
                controller,
                filename,
            )
            abort(status, message=msg)
```
I don't think it's worth the time/effort/space to decode three kinds of length errors; however, since you seem to value it, this code does return different status values for files which are too large and files which have bad (non-positive) lengths.

Also, I de-humanized the request's content length:  aside from the fact that it seems like `naturalsize()` doesn't handle negative numbers well, it will truncate the value to a single decimal place, and I think it's probably more important to display the actual value.